src/terminal/terminal-handlers.c

```diff
@@ -835,6 +835,28 @@ int guac_terminal_csi(guac_terminal* term, unsigned char c) {
 
                 break;
 
+            /* Scroll Up by amount */
+            case 'S':
+
+                /* Multiple linefeed */
+                for (int i = 0; i < argv[0]; i++)
+                    guac_terminal_linefeed(term, false);
+
+                term->char_handler = guac_terminal_echo;
```

Comment:
Why is this required? I don't see this assignment required in any of the other `case` statements - is there some reason why it needs to be assigned for these two new instances?

Reply:
Yes you are right, it is not useful since it is already assigned at line 1117 (in this branch).
I removed them.